test/integration/not-ready-test.js

```diff
@@ -0,0 +1,167 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var test = require('tape');
+var _ = require('underscore');
+var async = require('async');
+
+var testRingpopCluster = require('../lib/test-ringpop-cluster.js');
+var sendPing = require('../../lib/gossip/ping-sender.js');
+var sendPingReqs = require('../../lib/gossip/ping-req-sender.js');
+var Member = require('../../lib/membership/member.js');
+
+testRingpopCluster({
+    size: 4,
+    boostrapSize: 3,
+    tap: function(cluster) {
+        cluster[3].bootstrapHosts = ['127.0.0.1:12345', cluster[3].whoami()]
+    },
+    waitForConvergence: false
+}, 'not-ready error on ping', function t(bootRes, cluster, assert) {
+    assert.plan(1);
+
+    var ringpop = cluster[3];
+    cluster.forEach(function eachNode(node) {
+        node.membership.makeAlive(ringpop.whoami(), ringpop.incarnationNumber);
+    });
+    var unreachableMember = cluster[0].membership.findMemberByAddress(ringpop.whoami());
+
+    sendPing({
+        ringpop: cluster[0],
+        target: unreachableMember
+    }, function onPing(err, res) {
+        assert.equal(err.type, 'ringpop.not-ready');
+        assert.end();
+    });
+});
+
+testRingpopCluster({
+    size: 4,
+    boostrapSize: 3,
+    tap: function(cluster) {
+        cluster[3].bootstrapHosts = ['127.0.0.1:12345', cluster[3].whoami()]
+    },
+    waitForConvergence: false
+}, 'ping-req fails when not ready', function t(bootRes, cluster, assert) {
+    assert.plan(2);
+
+    var ringpop = cluster[3];
+    cluster.forEach(function eachNode(node) {
+        node.membership.makeAlive(ringpop.whoami(), ringpop.incarnationNumber);
+    });
+    var unreachableMember = cluster[0].membership.findMemberByAddress(ringpop.whoami());
+
+    sendPingReqs({
+        ringpop: cluster[0],
+        unreachableMember: unreachableMember,
+        pingReqSize: 2
+    }, function onPing(err, res) {
+        res.pingReqErrs.forEach(function each(pingReqRes) {
```

Comment:
That is somewhat tested in 'not-ready error on ping'. Any error is converted to a status error by the pinger and there is no sign of this in the response, so I don't see a trivial way yet.